pkg/server/metrics/metrics.go

```diff
@@ -99,7 +99,10 @@ func newServerMetrics() *ServerMetrics {
 			Name:      "ready_backend_connections",
 			Help:      "Number of konnectivity agent connected to the proxy server",
 		},
-		[]string{},
+		[]string{
+			"manager",
+			"idType",
+		},
```

Comment:
@logicalhan Is a change like this backward compatible, and if not is it more appropriate to create a new metric and deprecate the old?

Reply:
This is definitely not backwards compatible. This isn't in k8s-proper, so the same rules of API conformance do not necessarily hold here. I would at the very least add a release note clearly denoting the API change to the metric.

Reply:
Also what's the expected cardinality of manager and idType?

Reply:
I think cardinality is likely not a concern. (manager is currently 3, idType is about 3 or 4).

Reply:
> This is definitely not backwards compatible. This isn't in k8s-proper, so the same rules of API conformance do not necessarily hold here. I would at the very least add a release note clearly denoting the API change to the metric.

This is a binary we include in some distros of K8s (i.e. its an optional binary in K8s) so we should probably consider it part of K8s.

Reply:
> I think cardinality is likely not a concern. (manager is currently 3, idType is about 3 or 4).

Technically the idType is part of a CLI flag on the agent (Eg https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/addons/konnectivity-agent/konnectivity-agent-ds.yaml#L43), so theoretically its not bounded. 
Practically the number of managers and id types's are under the cloud provider's control. 
So I would expect this number to be small.

Reply:
See:
https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/cmd/agent/app/options/options.go#L178-L186

Reply:
Agreed. However that is on the agent side and the agent is at some level a reference implementation. 
If we had similar code on the server side to limit the cardinality of idType, then I think we could reasonably claim it was limited.